bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/overlay/FindReplaceOverlay.java

```diff
@@ -840,9 +855,13 @@ private void repositionTextSelection() {
 	}
 
 	private void updatePlacementAndVisibility() {
+		if (isInvalidTargetShell()) {
+			getShell().getDisplay().asyncExec(onParentShellChange);
```

Comment:
Instead of using the callback provided by the consumer, just internally updating the target seems to work for me:
```java
	getShell().getDisplay().asyncExec(() -> {
		close();
		setParentShell(targetPart.getSite().getShell());
		open();
		targetPart.setFocus();
	});
```
Can you try this and check whether it also works?

Reply:
![32](https://github.com/user-attachments/assets/425b4107-14df-4a42-b093-a30731fb6e0d)

I get this, which is close enough but not perfect. We need to make sure that launching the asynchronous function is idempotent. I can do that for example by setting a boolean flag once it is called and resetting it once the new overlay is actually opened.

I will do that in my office hourse